src/gui/MainWindow.cpp

```diff
@@ -208,12 +206,10 @@ MainWindow::MainWindow() :
 
 MainWindow::~MainWindow()
 {
-	for( QList<PluginView *>::iterator it = m_tools.begin();
-						it != m_tools.end(); ++it )
+	for( PluginView *view : m_tools )
 	{
-		Model * m = ( *it )->model();
-		delete *it;
-		delete m;
+		delete view;
+		delete view->model();
```

Comment:
This is likely going to segfault, as you're trying to use the member function `model()` of a deleted object.